src/zenml/model/model_config.py

```diff
@@ -30,24 +38,187 @@
+            description: The description of the model.
+            audience: The target audience of the model.
+            use_cases: The use cases of the model.
+            limitations: The known limitations of the model.
+            trade_offs: The tradeoffs of the model.
+            ethics: The ethical implications of the model.
+            tags: Tags associated with the model.
+            version: The model version name, number or stage is optional and points model context
+                to a specific version/stage. If skipped and `create_new_model_version` is False -
+                latest model version will be used.
+            version_description: The description of the model version.
+            create_new_model_version: Whether to create a new model version during execution
+            save_models_to_registry: Whether to save all ModelArtifacts to Model Registry,
+                if available in active stack.
+            delete_new_version_on_failure: Whether to delete failed runs with new versions for later recovery from it.
+            kwargs: Other arguments.
+        """
+        super().__init__(
+            name=name,
+            license=license,
+            description=description,
+            audience=audience,
+            use_cases=use_cases,
+            limitations=limitations,
+            trade_offs=trade_offs,
+            ethics=ethics,
+            tags=tags,
+            version=version,
+            version_description=version_description,
+            create_new_model_version=create_new_model_version,
+            save_models_to_registry=save_models_to_registry,
+            delete_new_version_on_failure=delete_new_version_on_failure,
+            model=kwargs.get("model", None),
+            model_version=kwargs.get("model_version", None),
+            user_not_yet_warned=kwargs.get("user_not_yet_warned", True),
+        )
+
+    class Config:
+        """Config class."""
+
+        smart_union = True
+
+    @root_validator
+    def _root_validator(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        """Validate all in one.
+
+        Args:
+            values: Dict of values.
+
+        Returns:
+            Dict of validated values.
+
+        Raises:
+            ValueError: If validation failed on one of the checks.
+        """
+        create_new_model_version = values.get(
+            "create_new_model_version", False
+        )
+        delete_new_version_on_failure = values.get(
+            "delete_new_version_on_failure", True
+        )
+        user_not_yet_warned = values.get("user_not_yet_warned", True)
+        if not delete_new_version_on_failure and not create_new_model_version:
+            if user_not_yet_warned:
+                logger.warning(
+                    "Using `delete_new_version_on_failure=False` and `create_new_model_version=False` has no effect."
+                    "Setting `delete_new_version_on_failure` to `True`."
+                )
```

Comment:
I don't even think this needs a warning, it's not like this is a dangerous configuration, one arg is simply not used, so why would users need to care?